jax/_src/numpy/lax_numpy.py

```diff
@@ -3847,6 +3847,36 @@ def diagflat(v, k=0):
   res = res.reshape(adj_length,adj_length)
   return res
 
+@_wraps(np.poly)
+def poly(seq_of_zeros):
```

Comment:
Please add a call to `_check_arraylike` on the input (we generally don't want to accept Python lists as inputs to JAX functions)

Reply:
My changes in https://github.com/google/jax/pull/6777/commits/30633e3839cef7c7d29a72a90d1aeee7b8d9c701 implement this suggestion.